### ISSUE TITLE
Deploy YunoHost v4.1.7.2 onto the µSD-card

### ISSUE THREAD

**Issue by frittro** [frittro/FMDS2022#1]

**Comment by frittro** [comment]
Instructions for burning Debian 10 (Buster) onto 32GB USB3.1 flash drive
==
Instruction source
====
https://wiki.debian.org/RaspberryPiImages

Instructions followed
====
```
$ export RPI_MODEL=4
$ lsblk
$ export SD_CARD=/dev/sdb
$ export DEBIAN_RELEASE=buster
$ cd ~/fmds2022/211226_yunohost-installfest/
$ wget https://raspi.debian.net/daily/raspi_${RPI_MODEL}_${DEBIAN_RELEASE}.img.xz
$ wget https://raspi.debian.net/daily/raspi_${RPI_MODEL}_${DEBIAN_RELEASE}.img.xz.sha256
$ sha256sum -c raspi_${RPI_MODEL}_${DEBIAN_RELEASE}.img.xz.sha256
$ xzcat raspi_${RPI_MODEL}_${DEBIAN_RELEASE}.img.xz | sudo dd of=${SD_CARD} bs=64k oflag=dsync status=progress
```
Output
====
```
1572601856 bytes (1.6 GB, 1.5 GiB) copied, 103 s, 15.3 MB/s
22270+5286 records in
22270+5286 records out
1572864000 bytes (1.6 GB, 1.5 GiB) copied, 103.024 s, 15.3 MB/s
```

**Comment by frittro** [comment]
# Installing YunoHost v4.1.7.2 on a Raspberry Pi 2, 3 or 4

## Instructions source
https://yunohost.org/en/install/hardware:rpi2plus
> NOTE: instructions marked with the ☢️ emoji have been custom added by @frittro to the default instructions for installing YunoHost, specifically for this particular project, and are not part of the default, recommended install procedure for YunoHost, as linked to above.

## Pre-requisites
* A Raspberry Pi 2, 3 or 4
* A power supply (either an adapter or a MicroUSB cable) for your board;
* A microSD card: 16GB capacity (at least), [class "A1"](https://en.wikipedia.org/wiki/SD_card#Class) highly recommended (such as [this SanDisk A1 card](https://www.amazon.fr/SanDisk-microSDHC-Adaptateur-homologu%C3%A9e-Nouvelle/dp/B073JWXGNT/));
* A [reasonable ISP](https://yunohost.org/en/isp), preferably with a good and unlimited upstream bandwidth
* An ethernet cable (RJ-45) to connect your server to your router.
* A computer to read this guide, flash the image and access your server.

## Download the YunoHost image
* [Download v4.1.7.2](https://build.yunohost.org/yunohost-buster-4.1.7.2-rpi-stable.img.zip)
* [Checksum](https://build.yunohost.org/yunohost-buster-4.1.7.2-rpi-stable.img.zip.sha256sum)
* [Signature](https://build.yunohost.org/yunohost-buster-4.1.7.2-rpi-stable.img.zip.sig)

## Flash the YunoHost image
If you are on GNU/Linux / macOS and know your way around command line, you may also flash your USB stick or SD card with `dd`. You can identify which device corresponds to your USB stick or SD card with `fdisk -l` or `lsblk`. A typical SD card name is something like `/dev/mmcblk0`. BE CAREFUL and make sure you got the right name.

Then run :
```bash
# Replace /dev/mmcblk0 if the name of your device is different...
dd if=/path/to/yunohost.img of=/dev/mmcblk0
```
## ☢️ **BACKUP**
> NOTE: until this project reaches the stage where a backup app has been installed under YunoHost, we will be using `rsync` instead to create backup image files. This is not ideal, and must be done from the laptop, with the µSD-card or USB thumb drive unmounted, to avoid data corruption.

> Instructions for using `rsync` to backup are [on the Raspberry Pi Stack Exchange](https://raspberrypi.stackexchange.com/a/5492/135415), thanks to [@goldilocks](https://raspberrypi.stackexchange.com/users/5538/goldilocks).

## Power up the board
* Plug the ethernet cable (one side on your main router, the other on your board).
* For advanced users willing to configure the board to connect to WiFi instead, see for example [here](https://www.raspberrypi.org/documentation/configuration/wireless/wireless-cli.md).
* Plug the SD card in your board
  * (Optional) You can connect a screen+keyboard directly on your board if you want to troubleshoot the boot process or if you're more comfortable to "see what happens" or want a direct access to the board.
* Power up the board
* Wait a couple minutes while the board autoconfigure itself during the first boot
* Make sure that your computer (desktop/laptop) is connected to the same local network (i.e. same internet box) as your server.

## ☢️ Update `sources.list`
The recent release of Debian 11 Bullseye has caused the sources for Debian 10 Buster to be moved from `stable` to `oldstable` in the repository, so your `sources.list` needs to be updated before you can update or upgrade the software of the underlying OS. You will need to send the command `sudo apt update` from the terminal soon after the first boot. See [this post on the YunoHost forums](https://forum.yunohost.org/t/updates-may-fail-because-repository-changed-its-suite-value-from-stable-to-oldstable/16931) for details.

## Proceed with the initial configuration
You can also perform the postinstallation with the command `yunohost tools postinstall` directly on the server, or [via SSH](https://yunohost.org/en/ssh).

### Main domain
This will be the domain used by your server's users to access the **authentication portal**. You can later add other domains, and change which one is the main domain if needed.
* If you're new to self-hosting and do not already have a domain name, we recommend using a **.nohost.me** / **.noho.st** / **.ynh.fr** (e.g. `homersimpson.nohost.me`). Provided that it's not already taken, the domain will be configured automatically and you won't need any further configuration step. Please note that the downside is that you won't have full-control over the DNS configuration.
* If you already own a domain name, you probably want to use it here. You will later need to configure DNS records as explained [here](https://yunohost.org/en/dns_config).

> Yes, you *have* to configure a domain name. If you don't have any domain name and don't want a **.nohost.me** / **.noho.st** / **.ynh.fr** either, you can set up a dummy domain such as `yolo.test` and tweak your `/etc/hosts` file such that this dummy domain points to the appropriate IP, as explained [here](https://yunohost.org/en/dns_local_network)).

###  Administration password
This password will be used to access to your server's administration interface. You will also use it to connect [via SSH](https://yunohost.org/en/ssh) or [SFTP](https://yunohost.org/en/filezilla) . In general terms, this is your **system's key**, choose it carefully!

## Create a first user
Once the postinstall is done, you should be able to actually log in the web admin interface using the administration password.

So far, your server now has an `admin` user - but `admin` is not a "regular" user and can't be used to log on [the user portal](https://yunohost.org/en/users).

> The first user you create is a bit special : it will receive emails sent to `root@yourdomain.tld` and `admin@yourdomain.tld`. These emails may be used to send technical informations or alerts.

Let's therefore add a first "regular" user.
```bash
yunohost user create johndoe
```

## Run the initial diagnosis
The diagnosis system is meant to provide an easy way to validate that all critical aspects of your server are properly configured - and guide you in how to fix issues. The diagnosis will run twice a day and send an alert if issues are detected.

> N.B. : **don't run away !** The first time you run the diagnosis, it is quite expected to see a bunch of yellow/red alerts because you typically need to configure DNS records (if not using a `.nohost.me`/`noho.st`/`ynh.fr` domain), add a swapfile if not enough ram and/or [port forwarding](https://yunohost.org/en/isp_box_config).

> If an alert is not relevant (for example because you don't intend on using a specific feature), it is perfectly fine to flag the issue as 'ignored' by going in the webadmin > Diagnosis, and clicking the ignore button for this specifc issue.
```bash
yunohost diagnosis run
yunohost diagnosis show --issues --human-readable
```

## Get a Let's Encrypt certificate
Once you configured DNS records and port forwarding (if needed), you should be able to install a Let's Encrypt certificate. This will get rid of the spooky security warning from earlier for new visitors.

For more detailled instructions, or to lean more about SSL/TLS certificates, see [the corresponding page here](https://yunohost.org/en/certificate).
```bash
yunohost domain cert-install
```

## Congratz!
You now have a pretty well configured server. If you're new to YunoHost, we recommend to have a look at [the guided tour](https://yunohost.org/en/overview). You should also be able to [install your favourite applications](https://yunohost.org/en/apps). Don't forget to [plan backups](https://yunohost.org/en/apps) !

## ☢️ **BACKUP**

**Comment by frittro** [comment]
# Preparing the µSD-card for YunoHost v4.1.7.2

1. Remove all other external devices from the laptop, just in case.
1. Insert the target µSD-card into the USB card reader, then insert that into the laptop's USB port.
1. Find the mountpoint of the µSD-card using the **`$ lsblk`** command. It is most likely to be `/dev/sdb`.
1. Wipe all existing data from the µSD-card using the command **`$ sudo dd if=/dev/zero of=/dev/sdb bs=4096 status=progress`** on the laptop.
1. On the laptop, use the command **`$ cd ~/dev/rpi/deployment/test-fmds.noho.st`** to change working directory.
1. Use **`$ wget  https://build.yunohost.org/yunohost-buster-4.1.7.2-rpi-stable.img.zip`** to download the YunoHost v4.1.7.2 package.
1. Use **`$ wget https://build.yunohost.org/yunohost-buster-4.1.7.2-rpi-stable.img.zip.sha256sum`** to download the checksum file.
1. Use **`$ wget https://build.yunohost.org/yunohost-buster-4.1.7.2-rpi-stable.img.zip.sig`** to download the signature file.
1. Verify the checksum with the command **`$ sha256sum -c yunohost-buster-4.1.7.2-rpi-stable.img.zip.sha256sum`**.
1. Validate the signature with the command **` `**.  (see [YNH forums](https://forum.yunohost.org/t/lack-in-installation-documentation-verify-integrity-and-signature/5799/3))
1. If the verification and validation pass, then burn the image to the µSD-card with the following commands...
   1. Confirm current working directory with **`$ cd ~/dev/rpi/deployment/test-fmds.noho.st`**.
   1. Confirm mountpoint of the µSD-card with **`$ lsblk`**.
   1. Unzip and write the image with **`$ unzip yunohost-buster-4.1.7.2-rpi-stable.img.zip |  sudo dd of=/dev/sdb bs=64k oflag=dsync status=progress`**
1. When writing is complete, unmount the µSD-card with the command **`$ udisksctl unmount -b /dev/sdb1 && udisksctl unmount -b /dev/sdb2 && udisksctl power-off -b /dev/sdb`**
1. Remove the USB card reader from the laptop's USB port, and remove the µSD-card from it.
1. Re-plug all other external devices back into the laptop.
